### DISCUSSION TITLE
page with multiple tags and post lists?

### DISCUSSION THREAD

**Discussion by atomGit** [GetPublii/Publii#1543] [Help]
i'm using the Simple v2.7 theme and i'd like to create a page like the tag pages (post title, thumb, excerpt), but with multiple, specific tags and without affecting any other tag/post pages...

```
                  tag a

post 1 title   post 2 title   post 3 title
thumbnail      thumbnail      thumbnail
excerpt        excerpt        excerpt

                  tag b

post 1 title   post 2 title   post 3 title
thumbnail      thumbnail      thumbnail
excerpt        excerpt        excerpt
```

i think the  ["How to display posts connected with a specific tag name?"](https://getpublii.com/dev/how-to-display-posts-connected-with-a-specific-tag-name/) guide is close to what i'm after, but it seems like that affects tag pages globally?

the css is no problem, but how would i create lists of posts for the given tags?

**Comment by candidexmedia** [comment]
Is this for the home page, or just a page/post on your site?

And if it's for a post, do you imagine re-using that logic for other posts/pages, or is this just a one-off thing?

**Comment by atomGit** [comment]
just an ordinary page, not the home page

**Reply by candidexmedia** [reply]
1. Create a [theme override](https://getpublii.com/dev/theme-overrides/) (optional, but recommended) using the text editor of your choice. I love VS Codium, but there are others.

2. Create a [post template](https://getpublii.com/dev/how-to-create-custom-templates/)

3. Edit your `post-NEW_TEMPLATE.hbs` (or whatever you decide to call it) and add the code in the guide your linked to earlier: https://getpublii.com/dev/how-to-display-posts-connected-with-a-specific-tag-name/. Start by adding the code where you'd like for it to be displayed in your page's structure. Use that code snippet for each tag you want to display, so:

```hbs
{{#each @website.contentStructure.tags}}
  {{#checkIf this.name '==' 'TAGNAMEA'}}
    {{#if this.postsNumber}}
    {{#each this.posts}}
      {{title}}
    {{/each}}
    {{else}}
      There are no posts connected with TAGNAMEA :(
    {{/if}}
  {{/checkIf}}
{{/each}}

{{#each @website.contentStructure.tags}}
  {{#checkIf this.name '==' 'TAGNAMEB'}}
    {{#if this.postsNumber}}
    {{#each this.posts}}
      {{title}}
    {{/each}}
    {{else}}
      There are no posts connected with TAGNAMEB :(
    {{/if}}
  {{/checkIf}}
{{/each}}
```

You should be able to use `this.name`, like the example, OR `this.id` to use the tag ID ([reference](https://getpublii.com/dev/tag-tags/)).

Inside of the...

```hbs
{{#each this.posts}}
      {{title}}
{{/each}}
```

...section is where you want to add your HTML + Handlebars markup that will be used for each post in the listing. Look at how this is formatted on the homepage and other areas of your site that have listings in your theme files. Consult the [Publii Handlebars tags reference for posts](https://getpublii.com/dev/post-tags/) and [featured images](https://getpublii.com/dev/featured-image-tags/) to know what `{{variable}}` to use for what. 

For example, you could do, for each tag (rough example based on your description):

```hbs
{{#each this.posts}}
      <h2>{{title}}</h2>
      <img src="{{feature-image-url-tag-double-check-in-docs-and-theme-files}}">
      <p>{{excerpt}}</p>
{{/each}}
```

4. Save your template. Restart Publii. Create a new post, and assign the template you just created as its template.

5. Preview the post to see how your listings are looking. Iterate, save, test, tweak, until you get your desired outcome.

Rather than create a whole new template, you could also use the [Theme Settings API](https://getpublii.com/dev/theme-settings-api/) to create an on and off toggle for all posts. You would then wrap the tag listing code under a condition which would only display the tag listings if the toggle is set to true. That said, I figured the template solution would be a little easier to explain and follow.

There's also probably a more efficient way of repeating the tags, but I'm no Handlebars JS pro lol

**Comment by atomGit** [comment]
> There's also probably a more efficient way of repeating the tags ...

yeah, i don't really want to hard-code tag names

i took a look at Simple's tag.hbs and tags.hbs thinking i might be able to incorporate code from the former into the latter and loop through the tags but i couldn't quickly get it to work

specifically i'd like to loop through only the "main tag" tags as set in the Simple theme

**Reply by candidexmedia** [reply]
> specifically i'd like to loop through only the "**main tag**" tags as set in the Simple theme

Should have specified that from the start. 😉There are forum post on this already: 

https://github.com/GetPublii/Publii/discussions/1401

https://github.com/GetPublii/Publii/discussions/1504 (for the opposite)

> i took a look at Simple's tag.hbs and tags.hbs thinking i might be able to incorporate code from the former into the latter and loop through the tags but i couldn't quickly get it to work

[Tag pages](https://getpublii.com/dev/tag-page-tags/) and the [Tags page](https://getpublii.com/dev/tags-page-tags/) have different page-specific Handlebars tags/variables and serve different functions. If your intent is to restrict the tags listed in the Tags page to just mainTags, you should edit the tags.hbs and consult the discussions linked to earlier.

If you want to keep your Tags page as is, *and* have a page strictly for mainTags listings, create a post template but use the mainTag code.

**Comment by bobmitro** [comment]
It can be handled via the **tags.hbs** file, try this snippet: 

```
{{#each tags}}
   <h2>{{this.name}}</h2>
   <ul>
      {{#getPostsByTags 5 this.id ""}}
         <li>
            <h3>{{ title }}</h3>
            <p>{{ excerpt }}</p>
         </li>
      {{/getPostsByTags}}
   </ul>
{{/each}}
```

**Reply by candidexmedia** [reply]
[getPostsByTags](https://getpublii.com/dev/getpostsbytags-helper/) is very helpful for efficiently iterating through particular tag IDs. 😁

What would the syntax be to only show mainTags? It sounds like OP doesn't want to hardcode the tag IDs to the template.

**Comment by atomGit** [comment]
i got stuck with `{{#getPostsByTags -1 this.id ""}}` - i need to either exclude a specific tag, or, preferably, include only published posts which are excluded from the blog page

i played with the newer syntax for `{{#getPostsByTags}}` using `concatenate` but i couldn't get it working
